### PR TITLE
First Backend version

### PR DESCRIPTION
**Pull request by donpaul999**
Using the HTML from Beatrice I've managed to add the backend for the application with JSP. 

Advantages:
-  we don't have to write redundant code, (ex: the footer & header for every page)
-  we can display the data sent by the web servlets.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/RecipeDetailsServlet.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/recipe")
+public class RecipeDetailsServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
```
Comment (bessmarche) [inline src/main/java/com/google/sps/servlets/RecipeDetailsServlet.java:33]: Add comment before function on line 29 explaining what doGet does (see style guideline)

File: src/main/java/com/google/sps/servlets/SearchServlet.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.List;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/search")
+public class SearchServlet extends HttpServlet {
+
```
Comment (bessmarche) [inline src/main/java/com/google/sps/servlets/SearchServlet.java]: Add comment before doGet explaining what it does

**Review by bessmarche — APPROVED** [review]

File: src/main/webapp/resources/style.css [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+.link:hover {
+  color: black;
+}
+
+#home-page {
+  align-items: center;
+  justify-content: center;
+}
+
+#welcome {
```
Comment (ramaroberto) [inline src/main/webapp/resources/style.css]: #welcome-message

File: src/main/webapp/resources/style.css [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+#searchbar {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 80%;
+}
+
+.buttonline {
```
Comment (ramaroberto) [inline src/main/webapp/resources/style.css]: .button-line

... or .button-online?

File: src/main/webapp/resources/style.css [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+}
+
+.btn {
```
Comment (ramaroberto) [inline src/main/webapp/resources/style.css:73]: Be consistent, it's either .button or .btn

**Review by ramaroberto — APPROVED** [review]
Ready to merge after solving comments!

File: src/main/webapp/resources/style.css (hunk 5 above)
Comment (donpaul999) [reply src/main/webapp/resources/style.css:73]: The class `btn` is from Bootstrap.
For consistency I renamed `button-line` class to `btn-line`

**Pull request merged**